








Blau syndrome SNP sets analysis, ENCODE genome annotation data
========================================================
P-values are corrected for multiple testing using FDR. 

For each analysis, only the significant enrichment results (p < 0.01) are shown. If an analysis does not have any significant results, it is not displayed at all. The analyses are separated by single lines (---).

Negative p-value indicates depletion.


```
## [1] "==============================================================="
## [1] "SNP set analyzed: data//TAB1-AA-1Encode"
## [1] "==============================================================="
## [1] "The total number of significantly ENRICHED associations is: 369"
## [1] "Top 10, or less, are shown"
##                                              Row.names Enrich.pval                                                                     V2
## 1              wgEncodeBroadHistoneH1hescH3k27me3StdPk   2.48e-165      H1-hESC H3K27me3 Histone Mods by ChIP-seq Peaks from ENCODE/Broad
## 2               wgEncodeBroadHistoneH1hescH3k4me1StdPk   1.32e-114       H1-hESC H3K4me1 Histone Mods by ChIP-seq Peaks from ENCODE/Broad
## 3               wgEncodeBroadHistoneH1hescSap3039731Pk   1.31e-135 H1-hESC SAP30 (39731) Histone Mods by ChIP-seq Peaks from ENCODE/Broad
## 4                wgEncodeBroadHistoneHsmmH3k27me3StdPk   1.85e-145         HSMM H3K27me3 Histone Mods by ChIP-seq Peaks from ENCODE/Broad
## 5                 wgEncodeBroadHistoneNhaH3k27me3StdPk   4.24e-143         NH-A H3K27me3 Histone Mods by ChIP-seq Peaks from ENCODE/Broad
## 6                  wgEncodeBroadHistoneNhdfadCtcfStdPk   1.64e-167          NHDF-Ad CTCF Histone Mods by ChIP-seq Peaks from ENCODE/Broad
## 7    wgEncodeSunyAlbanyGeneStGm12878Igf2bp1RbpAssocRna   4.95e-153                                                                   <NA>
## 8  wgEncodeSunyAlbanyGeneStGm12878Igf2bp1RbpAssocRnaV2   4.95e-153 GM12878 IGF2BP1 RBP Associated RNA by RIP-chip GeneST from ENCODE/SUNY
## 9         wgEncodeSunyAlbanyGeneStK562Celf1RbpAssocRna   2.92e-128                                                                   <NA>
## 10      wgEncodeSunyAlbanyGeneStK562Celf1RbpAssocRnaV2   2.92e-128      K562 CELF1 RBP Associated RNA by RIP-chip GeneST from ENCODE/SUNY
## [1] "---------------------------------------------------------------"
## [1] "The total number of significantly DEPLETED associations is: 132"
## [1] "Top 10, or less, are shown"
##                                              Row.names Enrich.pval                                                                       V2
## 1               wgEncodeBroadHistoneK562Ezh239875StdPk   -9.47e-57       K562 EZH2 (39875) Histone Mods by ChIP-seq Peaks from ENCODE/Broad
## 2                 wgEncodeBroadHistoneK562H3k9me1StdPk   -1.09e-77            K562 H3K9me1 Histone Mods by ChIP-seq Peaks from ENCODE/Broad
## 3                wgEncodeBroadHistoneK562H4k20me1StdPk   -1.74e-57           K562 H4K20me1 Histone Mods by ChIP-seq Peaks from ENCODE/Broad
## 4                       wgEncodeBroadHistoneK562NcorPk   -4.50e-53               K562 NCoR Histone Mods by ChIP-seq Peaks from ENCODE/Broad
## 5                    wgEncodeBroadHistoneK562P300StdPk   -8.70e-59               K562 P300 Histone Mods by ChIP-seq Peaks from ENCODE/Broad
## 6                       wgEncodeBroadHistoneK562PcafPk   -8.70e-59               K562 PCAF Histone Mods by ChIP-seq Peaks from ENCODE/Broad
## 7           wgEncodeGisChiaPetMcf7CtcfInteractionsRep2  -9.06e-172              MCF-7 CTCF ChIA-PET Interactions Rep 2 from ENCODE/GIS-Ruan
## 8           wgEncodeGisChiaPetMcf7Pol2InteractionsRep2   -2.42e-78              MCF-7 Pol2 ChIA-PET Interactions Rep 2 from ENCODE/GIS-Ruan
## 9    wgEncodeSunyAlbanyGeneStHelas3RipinputRbpAssocRna  -3.87e-102                                                                     <NA>
## 10 wgEncodeSunyAlbanyGeneStHelas3RipinputRbpAssocRnaV2  -3.87e-102 HeLa-S3 RIP-Input RBP Associated RNA by RIP-chip GeneST from ENCODE/SUNY
## [1] "---------------------------------------------------------------"
## [1] "==============================================================="
## [1] "SNP set analyzed: data//TAB1-AA-2other"
## [1] "==============================================================="
## [1] "The total number of significantly ENRICHED associations is: 42"
## [1] "Top 10, or less, are shown"
##        Row.names Enrich.pval                                                                                    V2
## 1        affyU95    8.67e-16                              Alignments of Affymetrix Consensus/Exemplars from HG-U95
## 2       ccdsGene    1.12e-24                                                                         Consensus CDS
## 3      dgvMerged    9.82e-15     Database of Genomic Variants: Structural Variant Regions (CNV, Inversion, In/del)
## 4  dgvSupporting    9.82e-15 Database of Genomic Variants: Supporting Structural Variants (CNV, Inversion, In/del)
## 5     fishClones    1.06e-25                                           Clones Placed on Cytogenetic Map Using FISH
## 6            gad    4.44e-20                         Genetic Association Studies of Complex Diseases and Disorders
## 7     jaxQtlAsIs    7.53e-89                                               MGI Mouse QTLs Coarsely Mapped to Human
## 8         rgdQtl    1.67e-23                                               Human Quantitative Trait Locus from RGD
## 9   simpleRepeat    1.93e-14                                                          Simple Tandem Repeats by TRF
## 10  ucscGenePfam    4.81e-18                                                            Pfam Domains in UCSC Genes
## [1] "---------------------------------------------------------------"
## [1] "The total number of significantly DEPLETED associations is: 10"
## [1] "Top 10, or less, are shown"
##               Row.names Enrich.pval                                                                       V2
## 1           all_bacends   -5.51e-26                                                                     <NA>
## 2  altSeqLiftOverPslP10   -2.34e-05                           GRCh37 Alternate Sequence Lift Over Alignments
## 3   altSeqLiftOverPslP9   -2.34e-05                                                                     <NA>
## 4      altSeqPatchesP10   -3.58e-05                                     Patches to GRCh37 Reference Sequence
## 5           laminB1Lads   -4.14e-04                         NKI LADs (Lamina Associated Domains, Tig3 cells)
## 6      lrgTranscriptAli   -3.10e-03               Locus Reference Genomic (LRG) Fixed Transcript Annotations
## 7          orfeomeGenes   -2.93e-18                                                                     <NA>
## 8           orfeomeMrna   -2.93e-18                                        ORFeome Collaboration Gene Clones
## 9              pubsBlat   -2.21e-35                        Sequences in Articles: PubmedCentral and Elsevier
## 10          pubsBlatPsl   -3.49e-23 Individual Sequence Matches of One Selected Article from Sequences Track
## [1] "---------------------------------------------------------------"
## [1] "==============================================================="
## [1] "SNP set analyzed: data//TAB1-AA-altSplicing"
## [1] "==============================================================="
## [1] "The total number of significantly ENRICHED associations is: 1"
## [1] "Top 10, or less, are shown"
##       Row.names Enrich.pval   V2
## 1 strangeSplice    2.94e-06 <NA>
## [1] "---------------------------------------------------------------"
## [1] "==============================================================="
## [1] "SNP set analyzed: data//TAB1-AA-coriellCNVs"
## [1] "==============================================================="
## [1] "The total number of significantly ENRICHED associations is: 1"
## [1] "Top 10, or less, are shown"
##                    Row.names Enrich.pval   V2
## 1 Chorionic_villus_cell_line   3.32e-130 <NA>
## [1] "---------------------------------------------------------------"
## [1] "The total number of significantly DEPLETED associations is: 2"
## [1] "Top 10, or less, are shown"
##      Row.names Enrich.pval   V2
## 1 B_Lymphocyte   -2.43e-25 <NA>
## 2   Fibroblast   -2.07e-83 <NA>
## [1] "---------------------------------------------------------------"
## [1] "==============================================================="
## [1] "SNP set analyzed: data//TAB1-AA-gapLocations"
## [1] "==============================================================="
## [1] "==============================================================="
## [1] "SNP set analyzed: data//TAB1-AA-genomicVariants"
## [1] "==============================================================="
## [1] "The total number of significantly ENRICHED associations is: 1"
## [1] "Top 10, or less, are shown"
##   Row.names Enrich.pval   V2
## 1      Loss    9.45e-31 <NA>
## [1] "---------------------------------------------------------------"
## [1] "The total number of significantly DEPLETED associations is: 4"
## [1] "Top 10, or less, are shown"
##     Row.names Enrich.pval   V2
## 1    Deletion   -1.39e-04 <NA>
## 2 Duplication   -2.72e-04 <NA>
## 3        Gain   -1.13e-25 <NA>
## 4   Inversion   -5.26e-12 <NA>
## [1] "---------------------------------------------------------------"
## [1] "==============================================================="
## [1] "SNP set analyzed: data//TAB1-AA-H3K4me3"
## [1] "==============================================================="
## [1] "The total number of significantly ENRICHED associations is: 15"
## [1] "Top 10, or less, are shown"
##                               Row.names Enrich.pval   V2
## 1        H3K4me3_Brain_Anterior_Caudate    7.79e-13 <NA>
## 2         H3K4me3_Brain_Cingulate_Gyrus    2.65e-07 <NA>
## 3      H3K4me3_Brain_Hippocampus_Middle    1.38e-09 <NA>
## 4  H3K4me3_Brain_Inferior_Temporal_Lobe    1.06e-12 <NA>
## 5        H3K4me3_Brain_Mid_Frontal_Lobe    9.33e-08 <NA>
## 6        H3K4me3_Brain_Substantia_Nigra    3.63e-08 <NA>
## 7             H3K4me3_CD3_Primary_Cells    1.64e-17 <NA>
## 8            H3K4me3_CD34_Primary_Cells    1.88e-06 <NA>
## 9       H3K4me3_CD8_Naive_Primary_Cells    2.83e-05 <NA>
## 10              H3K4me3_Skeletal_Muscle    2.20e-20 <NA>
## [1] "---------------------------------------------------------------"
## [1] "==============================================================="
## [1] "SNP set analyzed: data//TAB1-AA-histoneMarks"
## [1] "==============================================================="
## [1] "The total number of significantly ENRICHED associations is: 1"
## [1] "Top 10, or less, are shown"
##     Row.names Enrich.pval   V2
## 1 11_Weak_Txn    1.42e-03 <NA>
## [1] "---------------------------------------------------------------"
## [1] "The total number of significantly DEPLETED associations is: 2"
## [1] "Top 10, or less, are shown"
##          Row.names Enrich.pval   V2
## 1 13_Heterochromlo   -4.37e-05 <NA>
## 2 9_Txn_Transition   -6.37e-03 <NA>
## [1] "---------------------------------------------------------------"
## [1] "==============================================================="
## [1] "SNP set analyzed: data//TAB1-AA-ncRnas"
## [1] "==============================================================="
## [1] "==============================================================="
## [1] "SNP set analyzed: data//TAB1-AA-repeats"
## [1] "==============================================================="
## [1] "The total number of significantly ENRICHED associations is: 1"
## [1] "Top 10, or less, are shown"
##   Row.names Enrich.pval   V2
## 1      SINE    5.76e-14 <NA>
## [1] "---------------------------------------------------------------"
## [1] "==============================================================="
## [1] "SNP set analyzed: data//TAB1-AA-tfbsConserved"
## [1] "==============================================================="
```

```
## Error: no lines available in input
```

#```{r TAB1vsTAB2}
mtx <- merge(read.table(paste(dirs[1], "matrix.txt", sep="/"), sep="\t", row.names=1, header=T),
             read.table(paste(dirs[3], "matrix.txt", sep="/"), sep="\t", row.names=1, header=T),
             by="row.names")
rownames(mtx) <- mtx[, 1]; mtx <- mtx[, -1]               
showBarplot(mtx)
```

