<!DOCTYPE html>
<!-- saved from url=(0014)about:internet -->
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>
<meta http-equiv="x-ua-compatible" content="IE=9" >

<title>Blau syndrome SNP sets analysis, ENCODE genome annotation data</title>

<style type="text/css">
body, td {
   font-family: sans-serif;
   background-color: white;
   font-size: 12px;
   margin: 8px;
}

tt, code, pre {
   font-family: 'DejaVu Sans Mono', 'Droid Sans Mono', 'Lucida Console', Consolas, Monaco, monospace;
}

h1 { 
   font-size:2.2em; 
}

h2 { 
   font-size:1.8em; 
}

h3 { 
   font-size:1.4em; 
}

h4 { 
   font-size:1.0em; 
}

h5 { 
   font-size:0.9em; 
}

h6 { 
   font-size:0.8em; 
}

a:visited {
   color: rgb(50%, 0%, 50%);
}

pre {	
   margin-top: 0;
   max-width: 95%;
   border: 1px solid #ccc;
   white-space: pre-wrap;
}

pre code {
   display: block; padding: 0.5em;
}

code.r, code.cpp {
   background-color: #F8F8F8;
}

table, td, th {
  border: none;
}

blockquote {
   color:#666666;
   margin:0;
   padding-left: 1em;
   border-left: 0.5em #EEE solid;
}

hr {
   height: 0px;
   border-bottom: none;
   border-top-width: thin;
   border-top-style: dotted;
   border-top-color: #999999;
}

@media print {
   * { 
      background: transparent !important; 
      color: black !important; 
      filter:none !important; 
      -ms-filter: none !important; 
   }

   body { 
      font-size:12pt; 
      max-width:100%; 
   }
       
   a, a:visited { 
      text-decoration: underline; 
   }

   hr { 
      visibility: hidden;
      page-break-before: always;
   }

   pre, blockquote { 
      padding-right: 1em; 
      page-break-inside: avoid; 
   }

   tr, img { 
      page-break-inside: avoid; 
   }

   img { 
      max-width: 100% !important; 
   }

   @page :left { 
      margin: 15mm 20mm 15mm 10mm; 
   }
     
   @page :right { 
      margin: 15mm 10mm 15mm 20mm; 
   }

   p, h2, h3 { 
      orphans: 3; widows: 3; 
   }

   h2, h3 { 
      page-break-after: avoid; 
   }
}

</style>





</head>

<body>
<h1>Blau syndrome SNP sets analysis, ENCODE genome annotation data</h1>

<p>P-values are corrected for multiple testing using FDR. </p>

<p>For each analysis, only the significant enrichment results (p &lt; 0.01) are shown. If an analysis does not have any significant results, it is not displayed at all. The analyses are separated by single lines (&mdash;).</p>

<p>Negative p-value indicates depletion.</p>

<pre><code>## [1] &quot;===============================================================&quot;
## [1] &quot;SNP set analyzed: data//TAB1-AA-1Encode&quot;
## [1] &quot;===============================================================&quot;
## [1] &quot;The total number of significantly ENRICHED associations is: 369&quot;
## [1] &quot;Top 10, or less, are shown&quot;
##                                              Row.names Enrich.pval                                                                     V2
## 1              wgEncodeBroadHistoneH1hescH3k27me3StdPk   2.48e-165      H1-hESC H3K27me3 Histone Mods by ChIP-seq Peaks from ENCODE/Broad
## 2               wgEncodeBroadHistoneH1hescH3k4me1StdPk   1.32e-114       H1-hESC H3K4me1 Histone Mods by ChIP-seq Peaks from ENCODE/Broad
## 3               wgEncodeBroadHistoneH1hescSap3039731Pk   1.31e-135 H1-hESC SAP30 (39731) Histone Mods by ChIP-seq Peaks from ENCODE/Broad
## 4                wgEncodeBroadHistoneHsmmH3k27me3StdPk   1.85e-145         HSMM H3K27me3 Histone Mods by ChIP-seq Peaks from ENCODE/Broad
## 5                 wgEncodeBroadHistoneNhaH3k27me3StdPk   4.24e-143         NH-A H3K27me3 Histone Mods by ChIP-seq Peaks from ENCODE/Broad
## 6                  wgEncodeBroadHistoneNhdfadCtcfStdPk   1.64e-167          NHDF-Ad CTCF Histone Mods by ChIP-seq Peaks from ENCODE/Broad
## 7    wgEncodeSunyAlbanyGeneStGm12878Igf2bp1RbpAssocRna   4.95e-153                                                                   &lt;NA&gt;
## 8  wgEncodeSunyAlbanyGeneStGm12878Igf2bp1RbpAssocRnaV2   4.95e-153 GM12878 IGF2BP1 RBP Associated RNA by RIP-chip GeneST from ENCODE/SUNY
## 9         wgEncodeSunyAlbanyGeneStK562Celf1RbpAssocRna   2.92e-128                                                                   &lt;NA&gt;
## 10      wgEncodeSunyAlbanyGeneStK562Celf1RbpAssocRnaV2   2.92e-128      K562 CELF1 RBP Associated RNA by RIP-chip GeneST from ENCODE/SUNY
## [1] &quot;---------------------------------------------------------------&quot;
## [1] &quot;The total number of significantly DEPLETED associations is: 132&quot;
## [1] &quot;Top 10, or less, are shown&quot;
##                                              Row.names Enrich.pval                                                                       V2
## 1               wgEncodeBroadHistoneK562Ezh239875StdPk   -9.47e-57       K562 EZH2 (39875) Histone Mods by ChIP-seq Peaks from ENCODE/Broad
## 2                 wgEncodeBroadHistoneK562H3k9me1StdPk   -1.09e-77            K562 H3K9me1 Histone Mods by ChIP-seq Peaks from ENCODE/Broad
## 3                wgEncodeBroadHistoneK562H4k20me1StdPk   -1.74e-57           K562 H4K20me1 Histone Mods by ChIP-seq Peaks from ENCODE/Broad
## 4                       wgEncodeBroadHistoneK562NcorPk   -4.50e-53               K562 NCoR Histone Mods by ChIP-seq Peaks from ENCODE/Broad
## 5                    wgEncodeBroadHistoneK562P300StdPk   -8.70e-59               K562 P300 Histone Mods by ChIP-seq Peaks from ENCODE/Broad
## 6                       wgEncodeBroadHistoneK562PcafPk   -8.70e-59               K562 PCAF Histone Mods by ChIP-seq Peaks from ENCODE/Broad
## 7           wgEncodeGisChiaPetMcf7CtcfInteractionsRep2  -9.06e-172              MCF-7 CTCF ChIA-PET Interactions Rep 2 from ENCODE/GIS-Ruan
## 8           wgEncodeGisChiaPetMcf7Pol2InteractionsRep2   -2.42e-78              MCF-7 Pol2 ChIA-PET Interactions Rep 2 from ENCODE/GIS-Ruan
## 9    wgEncodeSunyAlbanyGeneStHelas3RipinputRbpAssocRna  -3.87e-102                                                                     &lt;NA&gt;
## 10 wgEncodeSunyAlbanyGeneStHelas3RipinputRbpAssocRnaV2  -3.87e-102 HeLa-S3 RIP-Input RBP Associated RNA by RIP-chip GeneST from ENCODE/SUNY
## [1] &quot;---------------------------------------------------------------&quot;
## [1] &quot;===============================================================&quot;
## [1] &quot;SNP set analyzed: data//TAB1-AA-2other&quot;
## [1] &quot;===============================================================&quot;
## [1] &quot;The total number of significantly ENRICHED associations is: 42&quot;
## [1] &quot;Top 10, or less, are shown&quot;
##        Row.names Enrich.pval                                                                                    V2
## 1        affyU95    8.67e-16                              Alignments of Affymetrix Consensus/Exemplars from HG-U95
## 2       ccdsGene    1.12e-24                                                                         Consensus CDS
## 3      dgvMerged    9.82e-15     Database of Genomic Variants: Structural Variant Regions (CNV, Inversion, In/del)
## 4  dgvSupporting    9.82e-15 Database of Genomic Variants: Supporting Structural Variants (CNV, Inversion, In/del)
## 5     fishClones    1.06e-25                                           Clones Placed on Cytogenetic Map Using FISH
## 6            gad    4.44e-20                         Genetic Association Studies of Complex Diseases and Disorders
## 7     jaxQtlAsIs    7.53e-89                                               MGI Mouse QTLs Coarsely Mapped to Human
## 8         rgdQtl    1.67e-23                                               Human Quantitative Trait Locus from RGD
## 9   simpleRepeat    1.93e-14                                                          Simple Tandem Repeats by TRF
## 10  ucscGenePfam    4.81e-18                                                            Pfam Domains in UCSC Genes
## [1] &quot;---------------------------------------------------------------&quot;
## [1] &quot;The total number of significantly DEPLETED associations is: 10&quot;
## [1] &quot;Top 10, or less, are shown&quot;
##               Row.names Enrich.pval                                                                       V2
## 1           all_bacends   -5.51e-26                                                                     &lt;NA&gt;
## 2  altSeqLiftOverPslP10   -2.34e-05                           GRCh37 Alternate Sequence Lift Over Alignments
## 3   altSeqLiftOverPslP9   -2.34e-05                                                                     &lt;NA&gt;
## 4      altSeqPatchesP10   -3.58e-05                                     Patches to GRCh37 Reference Sequence
## 5           laminB1Lads   -4.14e-04                         NKI LADs (Lamina Associated Domains, Tig3 cells)
## 6      lrgTranscriptAli   -3.10e-03               Locus Reference Genomic (LRG) Fixed Transcript Annotations
## 7          orfeomeGenes   -2.93e-18                                                                     &lt;NA&gt;
## 8           orfeomeMrna   -2.93e-18                                        ORFeome Collaboration Gene Clones
## 9              pubsBlat   -2.21e-35                        Sequences in Articles: PubmedCentral and Elsevier
## 10          pubsBlatPsl   -3.49e-23 Individual Sequence Matches of One Selected Article from Sequences Track
## [1] &quot;---------------------------------------------------------------&quot;
## [1] &quot;===============================================================&quot;
## [1] &quot;SNP set analyzed: data//TAB1-AA-altSplicing&quot;
## [1] &quot;===============================================================&quot;
## [1] &quot;The total number of significantly ENRICHED associations is: 1&quot;
## [1] &quot;Top 10, or less, are shown&quot;
##       Row.names Enrich.pval   V2
## 1 strangeSplice    2.94e-06 &lt;NA&gt;
## [1] &quot;---------------------------------------------------------------&quot;
## [1] &quot;===============================================================&quot;
## [1] &quot;SNP set analyzed: data//TAB1-AA-coriellCNVs&quot;
## [1] &quot;===============================================================&quot;
## [1] &quot;The total number of significantly ENRICHED associations is: 1&quot;
## [1] &quot;Top 10, or less, are shown&quot;
##                    Row.names Enrich.pval   V2
## 1 Chorionic_villus_cell_line   3.32e-130 &lt;NA&gt;
## [1] &quot;---------------------------------------------------------------&quot;
## [1] &quot;The total number of significantly DEPLETED associations is: 2&quot;
## [1] &quot;Top 10, or less, are shown&quot;
##      Row.names Enrich.pval   V2
## 1 B_Lymphocyte   -2.43e-25 &lt;NA&gt;
## 2   Fibroblast   -2.07e-83 &lt;NA&gt;
## [1] &quot;---------------------------------------------------------------&quot;
## [1] &quot;===============================================================&quot;
## [1] &quot;SNP set analyzed: data//TAB1-AA-gapLocations&quot;
## [1] &quot;===============================================================&quot;
## [1] &quot;===============================================================&quot;
## [1] &quot;SNP set analyzed: data//TAB1-AA-genomicVariants&quot;
## [1] &quot;===============================================================&quot;
## [1] &quot;The total number of significantly ENRICHED associations is: 1&quot;
## [1] &quot;Top 10, or less, are shown&quot;
##   Row.names Enrich.pval   V2
## 1      Loss    9.45e-31 &lt;NA&gt;
## [1] &quot;---------------------------------------------------------------&quot;
## [1] &quot;The total number of significantly DEPLETED associations is: 4&quot;
## [1] &quot;Top 10, or less, are shown&quot;
##     Row.names Enrich.pval   V2
## 1    Deletion   -1.39e-04 &lt;NA&gt;
## 2 Duplication   -2.72e-04 &lt;NA&gt;
## 3        Gain   -1.13e-25 &lt;NA&gt;
## 4   Inversion   -5.26e-12 &lt;NA&gt;
## [1] &quot;---------------------------------------------------------------&quot;
## [1] &quot;===============================================================&quot;
## [1] &quot;SNP set analyzed: data//TAB1-AA-H3K4me3&quot;
## [1] &quot;===============================================================&quot;
## [1] &quot;The total number of significantly ENRICHED associations is: 15&quot;
## [1] &quot;Top 10, or less, are shown&quot;
##                               Row.names Enrich.pval   V2
## 1        H3K4me3_Brain_Anterior_Caudate    7.79e-13 &lt;NA&gt;
## 2         H3K4me3_Brain_Cingulate_Gyrus    2.65e-07 &lt;NA&gt;
## 3      H3K4me3_Brain_Hippocampus_Middle    1.38e-09 &lt;NA&gt;
## 4  H3K4me3_Brain_Inferior_Temporal_Lobe    1.06e-12 &lt;NA&gt;
## 5        H3K4me3_Brain_Mid_Frontal_Lobe    9.33e-08 &lt;NA&gt;
## 6        H3K4me3_Brain_Substantia_Nigra    3.63e-08 &lt;NA&gt;
## 7             H3K4me3_CD3_Primary_Cells    1.64e-17 &lt;NA&gt;
## 8            H3K4me3_CD34_Primary_Cells    1.88e-06 &lt;NA&gt;
## 9       H3K4me3_CD8_Naive_Primary_Cells    2.83e-05 &lt;NA&gt;
## 10              H3K4me3_Skeletal_Muscle    2.20e-20 &lt;NA&gt;
## [1] &quot;---------------------------------------------------------------&quot;
## [1] &quot;===============================================================&quot;
## [1] &quot;SNP set analyzed: data//TAB1-AA-histoneMarks&quot;
## [1] &quot;===============================================================&quot;
## [1] &quot;The total number of significantly ENRICHED associations is: 1&quot;
## [1] &quot;Top 10, or less, are shown&quot;
##     Row.names Enrich.pval   V2
## 1 11_Weak_Txn    1.42e-03 &lt;NA&gt;
## [1] &quot;---------------------------------------------------------------&quot;
## [1] &quot;The total number of significantly DEPLETED associations is: 2&quot;
## [1] &quot;Top 10, or less, are shown&quot;
##          Row.names Enrich.pval   V2
## 1 13_Heterochromlo   -4.37e-05 &lt;NA&gt;
## 2 9_Txn_Transition   -6.37e-03 &lt;NA&gt;
## [1] &quot;---------------------------------------------------------------&quot;
## [1] &quot;===============================================================&quot;
## [1] &quot;SNP set analyzed: data//TAB1-AA-ncRnas&quot;
## [1] &quot;===============================================================&quot;
## [1] &quot;===============================================================&quot;
## [1] &quot;SNP set analyzed: data//TAB1-AA-repeats&quot;
## [1] &quot;===============================================================&quot;
## [1] &quot;The total number of significantly ENRICHED associations is: 1&quot;
## [1] &quot;Top 10, or less, are shown&quot;
##   Row.names Enrich.pval   V2
## 1      SINE    5.76e-14 &lt;NA&gt;
## [1] &quot;---------------------------------------------------------------&quot;
## [1] &quot;===============================================================&quot;
## [1] &quot;SNP set analyzed: data//TAB1-AA-tfbsConserved&quot;
## [1] &quot;===============================================================&quot;
</code></pre>

<pre><code>## Error: no lines available in input
</code></pre>

<p>#&ldquo;{r TAB1vsTAB2}
mtx &lt;- merge(read.table(paste(dirs[1], &quot;matrix.txt&rdquo;, sep=&ldquo;/&rdquo;), sep=&ldquo;\t&rdquo;, row.names=1, header=T),
             read.table(paste(dirs[3], &ldquo;matrix.txt&rdquo;, sep=&ldquo;/&rdquo;), sep=&ldquo;\t&rdquo;, row.names=1, header=T),
             by=&ldquo;row.names&rdquo;)
rownames(mtx) &lt;- mtx[, 1]; mtx &lt;- mtx[, -1]<br/>
showBarplot(mtx)</p>

<pre><code>
</code></pre>

</body>

</html>

