---
  bibliography: references.bib
csl: styles.ref/genomebiology.csl
output:
  word_document:
  reference_docx: styles.doc/NSF_grant_style.docx
html_document: default
---

# Project 1. 3D Epigenomics project

## Big picture

- When comparing 3D genomic structures, regions that are highly stable, or unstable, can be defined. 
- The question is, which epigenomic features are _differentially associated with_ those regions

## Goal

- Find/develop a method for testing associations of stable/unstable regions with biologically meaningful epigenomic regions. The significance, or strength, of the associations should be compared, and epigenomic features _differentially associated with_ the stable/unstable regions should be defined.
- Association as _enrichment_: In epigenomic enrichment analysis (https://www.ncbi.nlm.nih.gov/pubmed/27153607), a set of genomic regions is tested for enrichment in a set of epigenomic regions. 
    - Note: The resolution of 3D data is relatively coarse, so the 3D regions are large. Thus, they cannot be tested for enrichment in much smaller epigenomic regions. Investigate the length distribution of each set of regions
    - Tentative idea: Need to reverse the enrichment analysis - test whether a set of epigenomic regions is enriched in a set of 3D regions. `Lola` will work, just need to organize data and code, and run the analyses.
- Association as _proximity_ analysis: GenometriCorr, Stereogene

## Data

- Stable/unstable regions 
  - Dropbox/genes_analysis/gm12878_gm12878
    - gm12878_gm12878_rep_diff.bed
    - gm12878_gm12878_rep_stable.bed
    - whole_genome.bed
  - Dropbox/genes_analysis/k562
    - k562_diff_regions.bed
    - k562_stable_regions.bed
    - k562_whole_genome.bed

- Epigenomic data - GenomeRunner web server. LOLA data - Merlot, /home/sequencing/data/ExtData/LOLA
  - Stereogene
    - Subcompartments
    - Chrom states
  
## ToDo

- Stereogene
  - When the results are "0" or "nan", investigate why
  - What happens when no overlap?
  - Organize into barplots, sorted by the largest difference btw. stable/diff region

- Chromosome enrichment
  - Columns: Chr, p-val for Stable, p-val for Diff
  
- GenometriCorr
  - Organize projection results into barplots
  - Start on Absolute/relative distance tests 

## Timeline

- 20 work hours/week
- Meetings twice a week (tentatively, Tuesday/Friday 1pm). Prepare a 2-3 slides PPT presentation of the results and plans for next steps
- July 23. PowerPoing with a full set of results for a 50-min presentation
- August 1. Results are added to the manuscript draft
- August 15. Manuscript submission


# Project 2. Organizing epigenomic data.

## Big picture

- GenomeRunner (http://integrativegenomics.org/, https://github.com/mdozmorov/genome_runner/tree/shiny) is an enrichment analysis tool (https://www.ncbi.nlm.nih.gov/pubmed/27153607)
- It uses data from the ENCODE and Roadmap Epigenomics project
- The data are acquired, reformatted and organized into hierarchical folder structure using `dbcreator*` scripts (https://mdozmorov.github.io/grdocs/dbcreator/dbcreator.html)

## Goal

- The data is outdated. Need to update the database using the current implementation of `dbcreator*` scripts.
- The annotations for the data are currently organized semi-manually into an RData object (`gfAnnot.R` file in https://github.com/mdozmorov/R.genomerunner/tree/shiny). These annotations are critical for an understanding of the results. Need to develop a systematic method of obtaining the data using `dbcreator*` scripts AND creating the annotation for it. Need to make it flexible enough to allow an addition of other BED data files, listed in the https://github.com/mdozmorov/R.genomerunner/blob/shiny/GenomeRunner.xlsx, and their annotations.



# Project 3. 3D, or spatial enrichment analysis

## Big picture

- Genome form loops and recent technologies allow detecting interacting regions. Genes, even distant, can be in close proximity because they are co-regulated. 
- The data of how genomic regions interact with one another (think adjacency matrix) can be considered as a reference model
- Given a list of genes (or any genomic regions), how can we detect whether they fall within regions interacting more (or less) frequently that can be expected by chance?

## Goal

- Learn the method of 3D co-localization enrichment analysis developed as a part of HiBrowse (https://www.ncbi.nlm.nih.gov/pubmed/24511080)
- To be discussed, implement a 3D enrichment analysis of user-provided genomic regions in 3D chromatin interaction data.